drivers: spi: Fix SPI clock glitches at very high speed slew-rate

HoZHel opened the pull request:
Fix glitches on SPI clock line when the pin `slew-rate` is set to `very-high-speed`.

Under the following conditions:

- SPI master
- `spi-cpol` and `spi-hold-cs` properties are set.
- two consecutive SPI transactions are performed (i.e. two calls to SPI trasceive API) while keeping CS low

A glitch occurs on the SPI clock when the first transaction completes and in particular when the SPI is disabled.

This fix also applies to other values for slew-rate; however, at lower speeds the voltage drop is not significant to cause problem.

The figures below demonstrate the glitch captured by Saleae (digital analyzer) in both digital and analog modes before and after the fix.

![SCK_very-high-speed_fail](https://github.com/zephyrproject-rtos/zephyr/assets/135699308/2704a2bf-8116-4bec-9102-59119398df04)

![SCK_very-high-speed_pass](https://github.com/zephyrproject-rtos/zephyr/assets/135699308/76254421-905b-4377-a409-27d0a4986a3c)

msmttchr commented:
@HoZHel, I think it’s worth to consider another approach as described in https://docs.zephyrproject.org/latest/hardware/peripherals/spi.html#c.spi_release

See also #29320.

if I understand it correctly, when using `SPI_HOLD_ON_CS` and `SPI_LOCK_ON`, master should not reconfigure SPI between transactions and so disabling of SPI at the end of a transaction should not be needed.

msmttchr committed:
drivers: spi: keep SPI enabled if SPI_HOLD_ON_CS is used

Keep the SPI enabled when performing multiple transaction with
SPI_HOLD_ON_CS.
In such case, the end of transaction is marked by application calling
spi_release.

Signed-off-by: Michele Sardo <msmttchr@gmail.com>

msmttchr committed:
drivers: bluetooth: hci: Enable SPI_LOCK_ON mode

Use SPI_LOCK_ON to lock configuration for multiple transaction while CS is
kept low.

Change control of CS line from direct GPIO manipulation to SPI driver API.

Signed-off-by: Michele Sardo <msmttchr@gmail.com>

HoZHel commented:
@jhedberg  Would you please take a look at this PR?
